### PR TITLE
NOISSUE - Fix CI to fetch tags

### PR DESCRIPTION
**Pull request by dborovcanin**
# What type of PR is this?

This is a quick fix of CI scirpt.

## What does this do?

Since our code build includes the latest Git tag for the Health endpoint, we need to have at least the latest tag available during building binaries in the CI.

## Which issue(s) does this PR fix/relate to?

There is no such issue.

## Have you included tests for your changes?

N/A

## Did you document any new/modified feature?

N/A

### Notes

N/A

### CODE REVIEW

**Review by arvindh123 — DISMISSED** [review]

**Comment by dborovcanin** [comment]
Unfortunately, fetching tags on this action is still broken: https://github.com/actions/checkout/issues/1471. I had to add a manual step for that.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -16,6 +16,12 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
+        with:
```
Comment (rodneyosodo) [inline .github/workflows/build.yml]: Since this doesn't work do we still need it?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -28,6 +30,10 @@ jobs:
         with:
           version: v1.57.2
 
+      - name: Fetch tags for the build
```
Comment (rodneyosodo) [inline .github/workflows/tests.yml]: Shouldn't this only be needed on the build workflow?

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: Makefile [hunk 3]
```diff
@@ -12,7 +12,7 @@ DOCKERS = $(addprefix docker_,$(SERVICES))
 DOCKERS_DEV = $(addprefix docker_dev_,$(SERVICES))
 CGO_ENABLED ?= 0
 GOARCH ?= amd64
-VERSION ?= $(shell git describe --abbrev=0 --tags)
+VERSION ?= $(shell git describe --abbrev=0 --tags || echo 'unknown')
```
Comment (rodneyosodo) [inline Makefile]:
```suggestion
VERSION ?= $(shell git describe --abbrev=0 --tags 2>/dev/null || echo 'unknown')
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

**Review by rodneyosodo — APPROVED** [review]
LGTM

**Pull request merged**